Tutorials/Sprite-Related-Tutorials/Sprite Layering Tutorial/01-02-sprites-layering.mdx

```diff
@@ -47,56 +69,118 @@ With our code at this stage, we basically just have a single-layered `Sprite`. U
 layers to our first `Sprite` object for later use. For reference, if you'd like to confirm that the
 layer indexes are correct, or see for yourself for debugging / testing purposes, the function
-`SpriteAddLayer()` has a return type of an integer, which corresponds to the assigned layer index
-value, which can be wrapped as:
+[SpriteLayerCount()](https://splashkit.io/api/sprites/#sprite-layer-count) has a return type of an integer, which can be wrapped as:
 
+<Tabs>
+  <TabItem label="C#">
 ```csharp
 SplashKit.WriteLine(SplashKit.SpriteLayerCount(bodySprite));
 ```
+  </TabItem>
+  <TabItem label="C++">
+```c++
+write_line(sprite_layer_count(bodySprite));
+```
+</TabItem>
+</Tabs> 
+
+
+
 
 ## Showing and manipulating layers
 
 So now we have our created `Sprite`, our layers, and we can get to drawing our content to the
 screen, simple enough, right?
 
+<Tabs>
+  <TabItem label="C#">
 ```csharp
 SplashKit.DrawSprite(bodySprite);
 spriteLayers.Refresh();
 SplashKit.Delay(2000);
 ```
+  </TabItem>
+  <TabItem label="C++">
+```c++
+draw_sprite(bodySprite);
+refresh_window(current_window());
+delay(1000);
+```
+</TabItem>
+</Tabs> 
+
 
 And, our layers aren't showing. It's important to know that layers aren't shown by default, so in
 order to show our added layers to our initial `Sprite` object, we call a function called
-`SplashKit.SpriteShowLayer(Sprite, idx);` with our parameters. For us, this looks something like:
+[ShowSpriteLayer()](https://splashkit.io/api/sprites/#sprite-show-layer) with our parameters. For us, this looks something like:
 
+<Tabs>
+  <TabItem label="C#">
 ```csharp
 SplashKit.SpriteShowLayer(bodySprite, 1);
 SplashKit.DrawSprite(bodySprite);
 spriteLayers.Refresh();
 SplashKit.Delay(2000);
 ```
+  </TabItem>
+  <TabItem label="C++">
+```c++
+sprite_show_layer(bodySprite, 1);
+clear_screen(COLOR_WHITE);
+draw_sprite(bodySprite);
+refresh_window(current_window());
+delay(1000);
+```
+</TabItem>
+</Tabs> 
+
 
 Referencing the 1st layer (remember, 0th based indexing from initialization of a `Sprite` object),
 and then re-drawing and refreshing our screen will have the second layer show up as well. Copying
 this code for our third, our Man slowly puts on his cowboy hat and earns his sheriff badge.
 
 But what if you want to hide a layer afterwards? Maybe our Sheriff loses his hat, or his badge? It's
-simple enough, we can just call a `SplashKit.SpriteHideLayer(Sprite, idx)` function as follows:
+simple enough, we can just call a [SpriteHideLayer()](https://splashkit.io/api/sprites/#sprite-hide-layer) function as follows:
 
+<Tabs>
+  <TabItem label="C#">
 ```csharp
 SplashKit.SpriteHideLayer(bodySprite, 2);
 SplashKit.DrawSprite(bodySprite);
 spriteLayers.Refresh();
 SplashKit.Delay(2000);
 ```
+  </TabItem>
+  <TabItem label="C++">
+```c++
+SplashKit.SpriteHideLayer(bodySprite, 2);
+SplashKit.ClearScreen(Color.White);
+SplashKit.DrawSprite(bodySprite);
+spriteLayers.Refresh();
+SplashKit.Delay(2000);
```

Comment:
This C# code. Also same question as above RE: clearing screen. Should we be clearing the screen? If we are then we should add it to the C# code.